sgqlc/types/__init__.py

```diff
@@ -1529,7 +1529,7 @@ def __create_own_fields(cls):
         # that reports fields we're just deleting
         members = super(ContainerTypeMeta, cls).__dir__()
         for name in members:
-            if name.startswith('_') and (name not in ["_and" ,"_or" ,"_not" , "_eq" ,"_gt" ,"_gte" ,"_ilike" ,"_in" ,"_is_null" ,"_like" ,"_lt" ,"_lte" ,"_neq" ,"_nilike" ,"_nin" ,"_nlike" ,"_nsimilar" ,"_similar"]):
+            if name.startswith('_') and (name not in getattr(cls, '__field_names__', [])):
```

Comment:
I think this needs to be changed.

You can change the loop to be:

```py
def __create_own_fields(cls):
    for name in cls.__get_field_names():
        field = getattr(cls, name)

def __get_field_names(cls):
    try:
        return cls.__field_names__
    except AttributeError:
        members = super(ContainerTypeMeta, cls).__dir__()
        return (name  for  name in members if not name.startswith('_'))
```

This way we only `__dir__()` and filter if there are no declared fields.

Reply:
That doesn't to work since there seems to be some members that are obtained via `super(ContainerTypeMeta, cls).__dir__()` that are not added to `cls.__field_names__` by the codegen.

Reply:
On the other hand this seems work:
```
    def __get_field_names(cls):
        explicit_members = getattr(cls, "__field_names__", [])
        implicit_members = super(ContainerTypeMeta, cls).__dir__()
        return (
            name for name in implicit_members
            if (not name.startswith("_")) or (name in explicit_members)
        )
```

Reply:
I can't check right now, but I guess this is due the parent version, is that correct?

Like if you "implements Interface", you need to also get the inherited values. If this is the case, then I'd go with a different approach:
1. explicitly list everything in the final type, one single tuple to iterate
2. recurse into the parent class with `super(ContainerTypeMeta, cls).__get_field_names()` and concatenate the tuples.

Reply:
> I can't check right now, but I guess this is due the parent version, is that correct?
Not really sure what you mean by "version" here.

>2. recurse into the parent class with super(ContainerTypeMeta, cls).__get_field_names() and concatenate the tuples.

Tried this but then in fails in `delattr(cls, name)` at `__create_own_fields`. It sounds like `super(ContainerTypeMeta, cls).__get_field_names()` does some kind of filtering that prevents this from happening.

Reply:
I'll take a look at this tomorrow or the next day. I think I can make this work, otherwise I'll go with your version since it does the job.